### ISSUE TITLE
Dark mode optional

### ISSUE THREAD

**Issue by dsshard** [webpack/webpack-bundle-analyzer#642]
Hi! This is a super cool product, and often helps me out. Thank you!

I really miss the dark mode, since I often work at night, and the abrupt switch to the "light" tab is very irritating to the eyes.

I made a patch for myself that simply makes the body background #000 and it helps me. But it would be cool to add this as an option for the settings.

Thanks again for the product, you are awesome!

**Comment by dsshard** [comment]
https://github.com/webpack-contrib/webpack-bundle-analyzer/pull/643

**Comment by valscion** [comment]
Yeah a dark mode could be useful. I'd rather avoid adding a new option for it, though — but some sort of a UI feature for a dark mode could be useful.

I could imagine that it'd work like this:

1. Check [`localStorage`](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage) if a light/dark mode setting has been previously selected. If so, use that selection.
2. Check [`prefers-color-scheme`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme) and use that as a default
3. Allow user to override what `prefers-color-scheme` had and persist that to `localStorage`

The simplest first iteration could be to only add support for `prefers-color-scheme` and not mess with `localStorage` at all.

**Comment by Nitin-kumar-yadav1307** [comment]
Hi! I am a beginner and would like to work on implementing the optional dark mode feature as part of my first open-source contribution. Could you please assign this issue to me?

**Comment by evenstensberg** [comment]
Can't assign you but feel free to pick this up

**Comment by valscion** [comment]
Looks like it was possible to assign this issue to @Nitin-kumar-yadav1307 after all! I'll keep this one for you for a while, let us know if you hit obstacles @Nitin-kumar-yadav1307

**Comment by Nitin-kumar-yadav1307** [comment]
Thanks a lot for assigning this issue to me! .I’ve successfully set up the project locally and started implementing the Dark Mode toggle in the sidebar. I’ve added the feature and rebuilt the project  however, I can’t yet see the dark mode toggle reflected in the running app. I’m currently debugging to identify what might be preventing it from showing up. Please let me know if there’s any specific setup or component I should check to ensure the toggle renders correctly.

**Comment by evenstensberg** [comment]
Please don't use ai when communicating. Thanks.

**Comment by valscion** [comment]
@Nitin-kumar-yadav1307 I've unassigned you now since two weeks have passed and there hasn't been much new progress to show here.

So this issue is again up for grabs for anyone interested ☺️

**Comment by Nitin-kumar-yadav1307** [comment]
sorry , i don't know i have to show my progresses and i am stuck at a issuse but im still working on it, do i still work on it as not assignee?

**Comment by valscion** [comment]
You can but it's also possible that someone else beats you to it. But given that this issue has been open for more than a year now, it doesn't sound likely

**Comment by theEquinoxDev** [comment]
Hi, I'd love to work on this issue if it's still open. Could someone please confirm it's available?

**Comment by valscion** [comment]
Sure, go ahead @theEquinoxDev

**Comment by theEquinoxDev** [comment]
Thank you for the confirmation. I'll begin working on it and actively try to update you. I'll open a PR once everything is tested locally and is working fine.

**Comment by valscion** [comment]
This has been implemented in #683 and shipped in v5.1.0, thanks @theEquinoxDev! 🎉

**Comment by theEquinoxDev** [comment]
Thanks so much! Really glad this was useful. Looking forward to contributing more in future...
